### PR TITLE
[bitnami/mariadb-galera] new chart for MariaDB Galera Deployment

### PR DESCRIPTION
**Pull request by sameersbn**
**Description of the change**

This chart bootstraps a [MariaDB Galera](https://github.com/bitnami/bitnami-docker-mariadb-galera) cluster on [Kubernetes](http://kubernetes.io).

**Benefits**

MariiaDB Galera is a multi-master database cluster solution for synchronous replication and high availability.

**Possible drawbacks**

None 

**Applicable issues**

None 

**Additional information**

None

**Checklist** [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/mariadb-galera/templates/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+                  app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- else}}
+      {{- with .Values.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+      {{- end }}
+      {{- end }}
```
Comment (juan131) [inline bitnami/mariadb-galera/templates/statefulset.yaml]: Suggestion for affinity (using templates):
```yaml
      affinity:
        {{- with .Values.nodeAffinity }}
        nodeAffinity: {{ tpl (toYaml .) $ | nindent 10 }}
        {{- end }}
        {{- with .Values.podAffinity }}
        podAffinity: {{ tpl (toYaml .) $ | nindent 10 }}
        {{- end }}
        {{- if eq .Values.podAntiAffinity "hard" }}
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - topologyKey: "kubernetes.io/hostname"
            labelSelector:
              matchLabels:
                app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
                app.kubernetes.io/instance: {{ .Release.Name }}
        {{- else if eq .Values.podAntiAffinity "soft" }}
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 1
            podAffinityTerm:
              topologyKey: "kubernetes.io/hostname"
              labelSelector:
                matchLabels:
                  app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
                  app.kubernetes.io/instance: {{ .Release.Name }}
        {{- end }}
```

File: bitnami/mariadb-galera/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      {{- if .Values.nodeSelector }}
+      nodeSelector:
+        {{ toYaml .Values.nodeSelector | nindent 8 }}
+      {{- end -}}
+      {{- with .Values.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+      {{- end }}
```
Comment (juan131) [inline bitnami/mariadb-galera/templates/statefulset.yaml]: Evaluate nodeSelector and toleration as templates:
```yaml
      {{- with .Values.nodeSelector }}
      nodeSelector: {{ tpl (toYaml .) $ | nindent 8 }}
      {{- end }}
      {{- with .Values.tolerations }}
      tolerations: {{ tpl (toYaml .) $ | nindent 8 }}
      {{- end }}
```
Then edit the README.md so you add in the parameter description: "The value is evaluated as a template."

File: bitnami/mariadb-galera/templates/svc.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  {{- if and (eq .Values.service.type "LoadBalancer") .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+  {{ with .Values.service.loadBalancerSourceRanges }}
+{{ toYaml . | indent 4 }}
+{{- end }}
+  {{- end }}
+  {{- if and (eq .Values.service.type "ClusterIP") .Values.service.clusterIP }}
+  clusterIP: {{ .Values.service.clusterIP }}
+  {{- end }}
+  ports:
```
Comment (juan131) [inline bitnami/mariadb-galera/templates/svc.yaml]: Should gallera port (4567) exposed here too?

File: bitnami/mariadb-galera/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{{- else if contains "ClusterIP" .Values.service.type }}
+
+    kubectl port-forward --namespace {{ .Release.Namespace }} svc/{{ template "mariadb-galera.fullname" . }} {{ .Values.service.port }}:{{ .Values.service.port }} &
+    mysql -h 127.0.0.1 -P {{ .Values.service.port }} -u{{ if .Values.db.user }}{{ .Values.db.user }}{{ else }}root{{ end }} -p{{ if .Values.db.user }}$(kubectl get secret --namespace {{ .Release.Namespace }} {{ if .Values.existingSecret }}{{ .Values.existingSecret }}{{ else }}{{ template "mariadb-galera.fullname" . }}{{ end }} -o jsonpath="{.data.mariadb-password}" | base64 --decode){{ else }}$(kubectl get secret --namespace {{ .Release.Namespace }} {{ if .Values.existingSecret }}{{ .Values.existingSecret }}{{ else }}{{ template "mariadb-galera.fullname" . }}{{ end }} -o jsonpath="{.data.mariadb-root-password}" | base64 --decode){{ end }} {{ .Values.db.name }}
+
+{{- end }}
+
+To upgrade this helm chart:
+
+    helm upgrade --namespace {{ .Release.Namespace }} {{ .Release.Name }} stable/mariadb-galera \
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply bitnami/mariadb-galera/templates/NOTES.txt]: good catch!

File: bitnami/mariadb-galera/README.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+
+To uninstall/delete the `my-release` release:
+
+```bash
+$ helm delete --purge my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## LDAP
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply bitnami/mariadb-galera/README.md]: :+1:

File: bitnami/mariadb-galera/templates/svc.yaml (hunk 3 above)
Comment (sameersbn) [reply bitnami/mariadb-galera/templates/svc.yaml]: galera port 4567 is for internal communication between the galera pods. I believe we do not need to advertise this port to external services

File: bitnami/mariadb-galera/templates/statefulset.yaml (hunk 2 above)
Comment (sameersbn) [reply bitnami/mariadb-galera/templates/statefulset.yaml]: :+1:

File: bitnami/mariadb-galera/templates/statefulset.yaml (hunk 1 above)
Comment (sameersbn) [reply bitnami/mariadb-galera/templates/statefulset.yaml]: :+1:

**Comment by sameersbn** [comment]
updated the PR to address the review comments. PTAL

File: bitnami/mariadb-galera/templates/svc.yaml (hunk 3 above)
Comment (juan131) [reply bitnami/mariadb-galera/templates/svc.yaml]: Fair enough!

**Comment by sameersbn** [comment]
> @sameersbn something seems to be broken. Could you check it?

fixed

**Pull request merged**